src/BSON/ObjectID.c

```diff
@@ -352,16 +357,27 @@ static HashTable *php_phongo_objectid_get_gc(zval *object, phongo_get_gc_table t
 	*table = NULL;
 	*n = 0;
 
-	return zend_std_get_properties(object TSRMLS_CC);
+	return Z_OBJECTID_OBJ_P(object)->properties;
 } /* }}} */
 
-static HashTable *php_phongo_objectid_get_properties(zval *object TSRMLS_DC) /* {{{ */
+static HashTable *php_phongo_objectid_get_properties_hash(zval *object, bool is_debug TSRMLS_DC) /* {{{ */
 {
 	php_phongo_objectid_t *intern;
 	HashTable             *props;
 
 	intern = Z_OBJECTID_OBJ_P(object);
-	props = zend_std_get_properties(object TSRMLS_CC);
+
+	if (is_debug) {
+		ALLOC_HASHTABLE(props);
+		zend_hash_init(props, 1, NULL, ZVAL_PTR_DTOR, 0);
+	} else if (intern->properties) {
+		zend_hash_clean(intern->properties);
+		props = intern->properties;
+	} else {
+		ALLOC_HASHTABLE(props);
+		zend_hash_init(props, 1, NULL, ZVAL_PTR_DTOR, 0);
+		intern->properties = props;
+	}
```

Comment:
This section is done for every class. Is it not possible to extract this into its own function? It would save a lot of boiler plate.